espnet2/enh/espnet_enh_s2t_model.py

```diff
@@ -66,19 +77,72 @@ def forward(
                                       have the speech_lengths returned.
                                       see in
                                       espnet2/iterators/chunk_iter_factory.py
-            text: (Batch, Length) default None just to keep the argument order
-            text_lengths: (Batch,) default None for the same reason as speech_lengths
+            For Enh+ASR task:
+                text_spk1: (Batch, Length)
+                text_spk2: (Batch, Length)
+                ...
+                text_spk1_lengths: (Batch,)
+                text_spk2_lengths: (Batch,)
+                ...
+            For other tasks:
+                text: (Batch, Length) default None just to keep the argument order
+                text_lengths: (Batch,)
+                    default None for the same reason as speech_lengths
         """
-        if text_lengths is not None:
-            assert text_lengths.dim() == 1, text_lengths.shape
-        if speech_lengths is not None and text_lengths is not None:
+        if "text" in kwargs:
```

Comment:
Can we make it compact here? What do you think about the following:
```python
if any(key.startswith("text_spk") for key in kwargs.keys()):  # to check if there are text input using "text_spk*"
    filtered_keys = filter(...). # filter the keys of text_spk*
    num_refs = len(filtered_keys)
    text_ref_keys = [f"text_spk{i}" for i in range(num_refs)]
else:
    text_ref_keys = ["text"]

text = []
for key in text_ref_keys:
    text.append(kwargs.get(key, None))
    text_ref_lengths.append(kwargs.get(key + "lengths", None))

# remember to check the number of speakers is consistant with enhancement model
assert len(text) == self.enh_model.num_spk
```

Reply:
For the legacy case (`text` is used), the variables `text` and `text_ref_lengths` should be a tensor instead of a list. I feel it is a bit complicated to process both cases with unified code.

Reply:
For the legacy case (`text` is used), the variables `text` and `text_ref_lengths` should be a tensor instead of a list. I feel it is a bit complicated to process both cases with unified code.